java/com/engflow/notificationqueue/Client.java

```diff
@@ -89,10 +90,14 @@ public static void main(String[] args) throws Exception {
     } finally {
       if (channel != null) {
         channel.shutdownNow();
-        forwardChannel.shutdownNow();
+        if (shouldForwared) {
```

Comment:
Maybe check whether forwardChannel is not null here instead?

Reply:
Fixed!